### DISCUSSION TITLE
Using catch2 instead of google-test

### DISCUSSION THREAD

**Discussion by takev** [hikoworks/hikogui#404] [Ideas]
I've been thinking of using catch2 instead of google-test.

Catch2 has some interesting properties and I have noticed you can now add static_tests. I have not read anything about it, but it looks like you can write a single test that will be executed at both compile and run-time. Which is important to do if you use `if (std::is_contant_evaluated())`.

I am using google only because MSVC test-explorer only supports google-test and boost-test in cmake-mode.

Right now though I am thinking we that since test-explorer is almost always broken it doesn't matter that much.

**Comment by data-man** [comment]
https://github.com/boost-ext/ut (C++20) instead of Catch2. :)

**Reply by takev** [reply]
That looks cool.

Is there a way to get an `static_expect()` which will run the expression as a `static_assert()` but also leaves the expression in the test executable to be run at run-time?

**Reply by data-man** [reply]
Sorry, I don't know.
